Projects/LinkedListIterator/LinkedListIterator/LinkedListIterator/SLList.h

```diff
@@ -0,0 +1,374 @@
+#pragma once
+#include <functional>
+
+template <typename T>
+class TSLList
+{
+
+private:
+
+	struct FNode
+	{
+		T Element;
+		FNode* Next = nullptr;
+	};
+
+	FNode* Head = nullptr;
+	FNode* Tail = nullptr;
+
+	int Size = 0;
+
+public:
+
+	//-------------------------------------------------------------------------------------------
+	//								CONSTRUCTORS & DESTRUCTORS
+	//-------------------------------------------------------------------------------------------
+	TSLList()
+	{
+
+	}
+
+	//-------------------------------------------------------------------------------------------
+	TSLList(const T& FirstElement)
```

Comment:
I think the name could be a NewElement since it is a generator and can be reused